### PR TITLE
Add explicit authentication

### PR DESCRIPTION
**Pull request by pandaadb**
Hi,

I created this pull request because I ran into the following issue, documented here: 

https://github.com/elastic/elasticsearch-py/issues/470#issuecomment-256091032

It appears that python does not like the have hash signs in passwords when using the recommended  RFC-1738. So, for example, a password like "myPassword#" will break authentication with ES. 

This PR adds the following: 
- Option -a or --auth for authentication 

Elasticsearch will prefer RFC-style auth when both are specified. I could be convinced to add validation for this case (where both are specified). This version does not have that as I saw it unnecessary for my purposes :) 

Let me know your thoughts! 

-- Artur

### CODE REVIEW

File: es2csv.py [hunk 1]
```diff
@@ -264,6 +264,7 @@ def main():
     p.add_argument('-k', '--kibana_nested', dest='kibana_nested', action='store_true', help='Format nested fields in Kibana style.')
     p.add_argument('-r', '--raw_query', dest='raw_query', action='store_true', help='Switch query format in the Query DSL.')
     p.add_argument('-e', '--meta_fields', dest='meta_fields', action='store_true', help='Add meta-fields in output.')
+    p.add_argument('-a', '--auth', dest='auth', type=str, required=False, help='Elastic search basic authentication in the form of username:password')
```
Comment (taraslayshchuk) [inline es2csv.py]: Please add dot (.) at the end of sentence and please add documentation to readme about this option.

Comment (pandaadb) [inline es2csv.py]: Hi, thanks for considering the PR. I fixed the typos and added the option as well as an explicit example in your readme file. 

Thanks,

Artur

File: es2csv.py [hunk 2]
```diff
@@ -264,7 +264,7 @@ def main():
     p.add_argument('-k', '--kibana_nested', dest='kibana_nested', action='store_true', help='Format nested fields in Kibana style.')
     p.add_argument('-r', '--raw_query', dest='raw_query', action='store_true', help='Switch query format in the Query DSL.')
     p.add_argument('-e', '--meta_fields', dest='meta_fields', action='store_true', help='Add meta-fields in output.')
-    p.add_argument('-a', '--auth', dest='auth', type=str, required=False, help='Elastic search basic authentication in the form of username:password')
+    p.add_argument('-a', '--auth', dest='auth', type=str, required=False, help='Elasticsearch basic authentication in the form of username:password.')
```
Comment (taraslayshchuk) [inline es2csv.py]: Please move this arg between `-u` and `-i`, it should be better place for authentication.

File: README.rst [hunk 3]
```diff
@@ -104,6 +105,12 @@ With Authorization
 .. code-block:: bash
 
   $ es2csv -u http://login:password@my.cool.host.com:6666/es/ -q 'host: localhost' -o database.csv
+
+With explicit Authorization
+
+.. code-block:: bash
+
+  $ es2csv -u http://my.cool.host.com:6666/es/ -q 'host: localhost' -o database.csv -a login:password
```
Comment (taraslayshchuk) [inline README.rst]: Please move `-a login:password` in front of `-u`.

File: README.rst [hunk 4]
```diff
@@ -45,6 +45,7 @@ Usage
   -k, --kibana_nested                      Format nested fields in Kibana style.
   -r, --raw_query                          Switch query format in the Query DSL.
   -e, --meta_fields                        Add meta-fields in output.
+  -a, --auth                               Elasticsearch basic authentication in the form of username:password.
```
Comment (taraslayshchuk) [inline README.rst]:
```bash
-u, --url URL                                Elasticsearch host URL. Default is http://localhost:9200.
-a, --auth AUTH                              Elasticsearch basic authentication in the form of username:password.
-i, --index-prefixes INDEX [INDEX ...]       Index name prefix(es). Default is ['logstash-*'].
```
And please fix [header](https://github.com/taraslayshchuk/es2csv/pull/17/files/28735fe508445ac0e69eaa41b02022e04fd32d67..759b59964b5ba4949b0b9c19e5ae2e5f4523ef56#diff-88b99bb28683bd5b7e3a204826ead112R32):
`$ es2csv.py [-h] -q QUERY [-u URL] [-a AUTH] [-i INDEX [INDEX ...]]`

File: es2csv.py (hunk 2 above)
Comment (pandaadb) [inline es2csv.py]: Moved in the middle

File: README.rst (hunk 3 above)
Comment (pandaadb) [inline README.rst]: Moved to the beginning before u

File: README.rst (hunk 4 above)
Comment (pandaadb) [inline README.rst]: Added header args descirption

**Comment by taraslayshchuk** [comment]
Good job! 
Thanks for resolution this issue.

**Comment by pandaadb** [comment]
Yey :) No problem! 

Do you have an estimate on when you would release it?

**Comment by taraslayshchuk** [comment]
There is one PR #13 which almost complete and one issue #10 which I want try to fix.

**Pull request merged**